### ISSUE TITLE
Multi-thread trigger assertion

### ISSUE THREAD

**Issue by Flash-LHR** [meelgroup/bosphorus#38]
![image](https://user-images.githubusercontent.com/47357603/215394476-5687bcb6-d6b2-4891-b4a9-57e011fd065b.png)

The operation succeeds when using the default number of threads, but the assertion fails after setting the number of threads.
OS： 5.17.0-kali3-amd64
CPU：Intel(R) Core(TM) i5-8300H CPU @ 2.30GHz

[r05.cnf.txt](https://github.com/meelgroup/bosphorus/files/10534748/r05.cnf.txt)

**Comment by msoos** [comment]
Hi,

Thanks! Is this related to this? https://github.com/meelgroup/bosphorus/issues In other words, does that PR fix this bug or it causes this bug? I understand that this needs fixing, but want to understand first how to reproduce.

Cheers,

Mate

**Comment by Flash-LHR** [comment]
This issue has nothing to do with [pr](https://github.com/meelgroup/bosphorus/pull/39).

Pr only sets the number of threads of cryptominisat5, and the number of threads is always 1 regardless of whether the user specifies the thread parameter. In fact, I did not attempt to fix the above assertion failure.

**Comment by msoos** [comment]
Hi,

Sorry for the late response. It was indeed a bug! I fixed it in CryptoMiniSat, here:  https://github.com/msoos/cryptominisat/commit/9b4ee6f14849f7ea2b9efe2f33b698a3ba203947

If you recompile both CMS and Bosphorus (maybe no need to recompile Bosphorus, depending on your setup), it should now work! I have tested it :)

I would like to thank you for both the perfectly reproducible problem description and your patience. Sorry, lately I've been very busy so I couldn't get to this as soon as I hoped. Nevertheless, about 1 week to fix is probably not too bad. Thanks again for the bug report, it really helped,

Mate
